lib/ddtrace/contrib/patchable.rb

```diff
@@ -18,7 +18,7 @@ def present?
         end
 
         def compatible?
-          RUBY_VERSION >= '1.9.3' && present?
+          (RUBY_VERSION >= '1.9.3' || (defined?(JRUBY_VERSION) && JRUBY_VERSION >= '9.1.5')) && present?
```

Comment:
I think for resolving the original issue, this works just fine.

I would, however, like to suggest maybe that we should extract this out of `Contrib` (since this kind of compatibility is driven by the trace core), and into some kind of higher level function that can be used across the library.

My thought was maybe to extract it to `Datadog::Runtime` as:

```ruby
Datadog::Runtime.compatible?
```

What do you think? Open to other suggestions as well if there's a better way to represent this in the API.

Alternatively, part of me thinks this version check shouldn't be even necessary in the first place since the gemspec should exclude incompatible Rubies and JRubies from loading in the first place. Perhaps codifying this in the gemspec would be better, if possible.

Reply:
Yup, I think it makes sense to extract out of `Contrib` ... `Runtime` seems fine, something like?:
```
#dd-trace-rb/lib/ddtrace/runtime.rb
module Datadog
  module Runtime
    def self.compatible?
      RUBY_VERSION >= '1.9.3' || (defined?(JRUBY_VERSION) && JRUBY_VERSION >= '9.1.5')
    end
  end
end
```
And then require that in `Datadog::Contrib`? Can push up a change like that and some specs around it

Also fwiw there's some similar helper methods floating around the repo as well, opentracing has a `.supported?` method as well with slightly different requirements, not sure how to handle that.

Regarding the gemspec i am not sure what configuration would look like as I don't believe the `spec.required_ruby_version` would support a config option for jruby, I have to look into the gemspec docs a bit more to be honest here.

Reply:
Okay, if we can somehow rig something into the gemspec, that would be ideal.

Short of that, I think what you have above looks good. Maybe we rename it to `supported?` such that it's congruent with OpenTracing.

Reply:
So i pulled this out of `Contrib` and moved it to Datadog::Runtime, generally though the more I look into this the more I doubt my original approach, that checking for `JRUBY_VERSION` would resolve warn messages the user was getting on this open issue. I apologize if this ended up being an unnecessary rabbit hole, as it looks like JRuby would still define `RUBY_VERSION` in line with our expectations, testing this locally jruby with jruby 9.2.6 a bit

```
COMP10929:ddtrace eric.mustin$ jruby --version
jruby 9.2.6.0 (2.5.3) 2019-02-11 15ba00b Java HotSpot(TM) 64-Bit Server VM 12.0.1+12 on 12.0.1+12 +jit [darwin-x86_64]
COMP10929:ddtrace eric.mustin$ jirb
jruby-9.2.6.0 :001 > RUBY_VERSION
 => "2.5.3" 
jruby-9.2.6.0 :002 > JRUBY_VERSION
 => "9.2.6.0" 
jruby-9.2.6.0 :003 > RUBY_PLATFORM
 => "java" 
```

So, tl;dr while I think having a higher level `supported?` check is useful, i'm not sure if this really solves anything for the user's issue

Reply:
Would you not recommend making this change then?

Reply:
Yea, unfortunately I don't have enough confidence that this is providing anything useful at this point, although eventually it may make sense to extract some of this logic into `Datadog::Runtime`

Reply:
Ah, okay. Then I guess we should close this PR.